distributed/tests/test_worker.py

```diff
@@ -865,6 +865,12 @@ async def test_worker_dir(c, s, a, b):
         test_worker_dir()
 
 
+@gen_cluster(nthreads=[])
+async def test_false_worker_dir(s):
+    async with Worker(s.address, loop=s.loop, local_directory=""):
```

Comment:
```suggestion
    async with Worker(s.address, local_directory=""):
```

Specifying `loop=` isn't necessary if you're in the same event loop for a Worker.  Really, we should consider removing the keyword.